### PR TITLE
Docker Compose Enhancements

### PR DESCRIPTION
**Pull request by l-monninger**
# Summary
- **RFCs**: $\emptyset$.
- **Categories**: `protocol-units`

Small improvements to docker compose to include container testing. 

# Changelog

# Testing

1. e2e container tests added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +50,28 @@ just m1-da-light-node native build.setup.test.local
 just monza-full-node native build.setup.test.local
 ```
 
+## Run with Docker Compose
+When running with `docker compose` specif your revision in a file `.env` at the root of the project. The file should look like this:
+
+```bash
+REV=0fe2a4f28820c04ca0db07cdd44cafc98b792f3f
+```
```
[the remark this answers is not in the document]
Comment (l-monninger) [reply README.md:58]: Right now, we don't have a `latest` flow for most delivery; we're pinning by commit hash. Doing it that way, while more cumbersome, is more deterministic. Perhaps semver to come. 

We do have branch specific tags, if you look at the packages on this repo, you'll see them. 

But, yes, eventually latest tags will be introduced for people just looking to play around.

[the remark this answers is not in the document]
Comment (l-monninger) [reply README.md:58]: > Perhaps it's better to put this env file under ./docker/compose to make its purpose clear to somebody who stumbles upon it in the repository. Or if it must be in the root, clarify the variable's purpose like CONTAINER_REV.

I think CONTAINER_REV makes the most sense to me. I will make the change.

**Comment by l-monninger** [comment]
I believe what matters here is the underlying Docker Engine API: https://github.com/docker/compose/pull/11680

So, I would essentially need to actually install Docker 26 on the runner

File: util/godfig/src/backend/mod.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+pub(crate) trait BackendInternalOperations {
```
Comment (mzabaluev) [inline util/godfig/src/backend/mod.rs:1]: This looks out of place; an accidental commit from the Godfig work?

**Review by mzabaluev — APPROVED** [review]
Looks good! One spuriously added file notwithstanding.

File: util/godfig/src/backend/mod.rs (hunk 2 above)
Comment (l-monninger) [reply util/godfig/src/backend/mod.rs:1]: From `Godfig`

**Pull request merged**